lib/agent/actions/fileretrieval/index.js

```diff
@@ -17,55 +19,84 @@ var fs = require('fs'),
 var common = require('./../../common'),
     system = common.system,
     node_bin = path.join(system.paths.current, 'bin', 'node'),
-    run_as_user = common.system.run_as_user,
     logger = common.logger,
     os_name = process.platform.replace('darwin', 'mac').replace('win32', 'windows');
 
 var em,
     cp;
 
+var UPLOAD_SERVER = 'http://solid.preyproject.com/upload/upload';
+
 exports.check_pending_files = function() {
   files.run_stored();
 }
 
 exports.start = function(options, cb) {
   var file_path = options.path,
-      user = options.user,
-      file_id = options.file_id;
+      file_id = options.file_id,
+      file_size = parseInt(options.size),
+      user = options.user;
 
   em = em || new EventEmitter();
   if (cb) cb(null, em);
 
-  files.store(file_id, file_path, user);
-  files.show();
+  if (!options.resumable)
+    files.store(file_id, file_path, file_size, user);
+
+  var file = {
+    total: 0,
+    path: file_path,
+    user: user,
+    id: file_id,
+    size: file_size,
+    resumable: options.resumable
+  }
 
-  retrieve_file(file_path, user, file_id, function() {
-    files.del(file_id);
+  retrieve_file(file, function() {
+    files.del(file.id);
   });
 }
 
-function retrieve_file(file_path, user, file_id, cb) {
-  // We need to run the tree walker script personifying
-  // the owner of the corresponding /Users or /home subdirectories
-  if (os_name !== 'windows' && !user) {
-    return em.emit('error', new Error('Options should specify user.'));
+function retrieve_file(file, cb) {
+  if (file.resumable) {
+    var url = UPLOAD_SERVER + '?uploadID=' + file.id;
```

Comment:
add a comment describing the protocol (for the community), saying that this call basically gets the last byte processed by the upload server in order to resume the upload from that position